service/interpreter/timerProcessor.go

```diff
@@ -144,3 +150,24 @@ func (t *TimerProcessor) AddTimers(stateExeId string, commands []iwfidl.TimerCom
 	}
 	t.stateExecutionCurrentTimerInfos[stateExeId] = timers
 }
+
+// FixTimerCommandFromActivityOutput converts the durationSeconds to firingUnixTimestampSeconds
+// doing it right after the activity output so that we don't need to worry about the time drift after continueAsNew
+func FixTimerCommandFromActivityOutput(now time.Time, request iwfidl.CommandRequest) iwfidl.CommandRequest {
+	var timerCommands []iwfidl.TimerCommand
+	for _, cmd := range request.GetTimerCommands() {
+		if cmd.HasDurationSeconds() {
+			timerCommands = append(timerCommands, iwfidl.TimerCommand{
+				CommandId:                  cmd.CommandId,
+				FiringUnixTimestampSeconds: iwfidl.PtrInt64(now.Unix() + int64(cmd.GetDurationSeconds())),
+			})
+		} else {
+			timerCommands = append(timerCommands, iwfidl.TimerCommand{
+				CommandId:                  cmd.CommandId,
+				FiringUnixTimestampSeconds: cmd.FiringUnixTimestampSeconds,
+			})
```

Comment:
Maybe there's a technical reason for this but why don't we just append the original `cmd` here (like `timerCommands = append(timerCommands, cmd)`)?

Reply:
Yeah I think that's simpler to do that. Thanks for pointing it out :D